CODING_STANDARDS.md

```diff
@@ -0,0 +1,378 @@
+```
+
+Bad:
+
+```tsx
+setTimeout(doSomething, 86400000);
+```
+
+### __Comments__
+
+Always use multiline comments instead of single line comments
+
+_Multiline Good:_
+
+```tsx
+/** Comment */
+```
+
+_Single Line Bad:_
+
+```tsx
+// Comment
+```
+
+### TypeScript issues
+
+If there is a TypeScript issue which cannot be resolved because it is coming from another library or will be fixed in the future, utilize the annotation `@ts-expect-error` instead of `@ts-ignore` as this will inform us when the TypeScript issue is resolved and not be left stale in the code
+
+### __Literals__
+
+#### __Objects__
+
+- Object should always be created by `{}` and not `new Object()`
+- When creating an empty Object, use `{}` without spaces
+- When using an Object with keys
+   - Keys should be sorted alphabetically (case-insensitive) as this is a logical way to read a list
+   - If this is a type declaration, required types should appear alphabetically before optional types
+
+Good:
+
+```tsx
+const obj = { A: 1, b: 2, C: 3 };
+
+const obj = {
+    A: 1,
+    b: 2,
+    C: 3,
+};
+
+type Type = {
+    requiredTypeOne: string;
+    requiredTypeTwo: number;
+    optionalTypeOne?: string;
+    optionalTypeTwo?: number;
+};
+```
+
+Bad:
+```tsx
+const obj = {A:1,b:2,C:3};
+
+const obj = {A:1, b:2, C:3};
+
+const obj = {A : 1, b : 2, C : 3};
+
+const obj = { "A" : 1, "b" : 2, "C" : 3 };
+
+const obj = { A : 1, b : 2, C : 3 };
+
+const obj = { A :1, b :2, C :3 };
+
+const obj = { A : 1 , b : 2 , C : 3 };
+
+const obj = {
+    A : 1,
+    b : 2,
+    C : 3,
+};
+
+
+const obj = {
+    A : 1
+  , b : 2
+  , C : 3
+};
+
+type Type = {
+    optionalTypeOne?: string;
+    requiredTypeOne: string;
+    optionalTypeTwo?: number;
+    requiredTypeTwo: number;
+};
+```
+
+#### __Arrays__
+
+- Arrays should always be created by `[]` and not `new Array()`.
+- When creating an empty Array, use `[]` without spaces.
+- When creating an Array with values
+   - The comma should have no space before and 1 space behind itself
+   - The last value shouldn't be trailed with a comma on a single line but should on a multiline
+   - The comma shouldn't start a new line. It should always trail the previous value
+
+Good:
+```tsx
+const arr = [1, 2, 3];
+
+const arr = [
+    1,
+    2,
+    3,
+];
+```
+
+Bad:
+```tsx
+const arr = [1,2,3];
+
+const arr = [ 1, 2, 3 ];
+
+const arr = [ 1 , 2 , 3 ];
+
+const arr = [
+    1, 
+    2, 
+    3
+];
+
+const arr = [
+    1
+  , 2
+  , 3 
+];
+```
+
+### __If statements__
+
+- Avoid single line `if` statements when possible
+- Opening parenthesis should not have a space after it but should have a space between it and the `if` syntax
+- Closing parenthesis should not have a space before it but should have a space after it and before the open bracket `{` syntax
+
+
+### __JSX__
+
+Always use `()` surrounding the components in the return so the tags are not offset when auto-formatting the indentations
+
+### __Blocks/if statements__
```

Comment:
This should be forced by eslint and prettier?

Reply:
for the most part yes